src/Umbraco.Core/Routing/UmbracoRequestPaths.cs

```diff
@@ -130,5 +128,21 @@ public bool IsClientSideRequest(string absPath)
             var ext = Path.GetExtension(absPath);
             return !ext.IsNullOrWhiteSpace();
         }
+		
+        /// <summary>
+        /// Checks if the path is from a PluginController route.
+        /// </summary>
+        private bool IsPluginControllerRoute(string path)
+        {
+            var count = 0;
+            foreach (var c in path)
+            {
+                if (Array.FindIndex(Constants.CharArrays.ForwardSlash, x => x == c) != -1 && ++count >= 3)
```

Comment:
This searches an array of length one and then captures c in the predicate. Perhaps this could be written as `if(c == '/' && ++count >=3)`

Reply:
I agree, I don't understand why that's an array at all, but I tried to keep it as-is.